.cirrus.yml

```diff
@@ -29,9 +29,10 @@ task:
     MAVEN_OPTS: "-Dhttp.keepAlive=false \
       -Dmaven.repo.local=%CIRRUS_WORKING_DIR%/.m2 \
       -Dmaven.wagon.http.retryHandler.count=3"
-    MAVEN_VERSION: "3.8.4"
+    # This Maven version here should be same as minimum defined maven version in pom.xml
```

Comment:
@relentless-pursuit attention to your own comment :)

Reply:
Thanks for pointing it out. The issue says minimum version be 3.5.4 and the version which possibly handles the CI failure is 3.6.1. Shall i remove the comment?

Reply:
>Let's find when this parameter was added and let's bump minimum version to it.

@romani to be clear: you are good to update project minimum version to 3.6.1 (pom.xml, documentation)? This version is almost five years old at this point: https://maven.apache.org/docs/history.html

Reply:
> > Let's find when this parameter was added and let's bump minimum version to it.
> 
> @romani to be clear: you are good to update project minimum version to 3.6.1 (pom.xml, documentation)? This version is almost five years old at this point: https://maven.apache.org/docs/history.html, it should be fine.

the CI was fialing with 3.6.1:
```
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:build-helper-maven-plugin:3.5.0:add-source (add-source) on project checkstyle: The plugin org.codehaus.mojo:build-helper-maven-plugin:3.5.0 requires Maven version 3.6.3 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:build-helper-maven-plugin:3.5.0:add-source (add-source) on project checkstyle: The plugin org.codehaus.mojo:build-helper-maven-plugin:3.5.0 requires Maven version 3.6.3
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:172)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginIncompatibleException: The plugin org.codehaus.mojo:build-helper-maven-plugin:3.5.0 requires Maven version 3.6.3
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.checkRequiredMavenVersion (DefaultMavenPluginManager.java:309)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:168)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginIncompatibleException
validation failed with error code #1.
```

Reply:
`build-helper-maven-plugin` version `3.5.0` requires minimum version to be `3.6.3`

Reply:
Min version in pom should be any version that shown at error like `Caused by: org.apache.maven.plugin.PluginIncompatibleException: The plugin org.codehaus.mojo:build-helper-maven-plugin:3.5.0 requires Maven version 3.6.3`

To fail build quickly and be resolved quickly by user.

I never think on such version, if some plugin complains, we bump our min version up up to min required by plugins we use .

Reply:
Please use `3.6.3`.